plugins/inputs/elasticsearch/elasticsearch.go

```diff
@@ -221,7 +241,7 @@ func (e *Elasticsearch) Gather(acc telegraf.Accumulator) error {
 				}
 			}
 
-			if e.ClusterStats && (e.isMaster || !e.ClusterStatsOnlyFromMaster || !e.Local) {
+			if e.ClusterStats && (e.serverInfo[s].isMaster() || !e.ClusterStatsOnlyFromMaster || !e.Local) {
```

Comment:
Check this out, there is a way to do this in the request itself in >=6.3.  Unfortunately, I think we still want support for older versions.

https://www.elastic.co/guide/en/elasticsearch/reference/6.8/cluster.html#cluster-nodes

Reply:
I would prefer getting this merged first (as this will work for all versions).
My next pull req would be adding 6+ support, and will check to implement this then :)

Reply:
Yeah, unfortunately I don't think we can really use this at all until we drop ES5 support.